13_xml_document/xml_document.rb

```diff
@@ -0,0 +1,40 @@
+class XmlDocument
+  def initialize(pretty=false)
+    @xml = ""
+    @pretty = pretty
+    @level = 0
+  end
+
+  def method_missing(method, attributes = {}, &block)
+    @xml = ""
+    tag = method.to_s
+ 
+    append_tag_to_xml(tag, attributes)
+
+    if block_given?
+      indent if @pretty
+      @xml += "#{yield}</#{tag}>"
+      indent(true) if @pretty
+    else
+      format_to_self_closing
+    end
+    @xml
```

Comment:
See in https://trello.com/c/grccZlGR/4-design-patterns the `Builder` pattern. You are constructing the string here, and you can do it better and avoid replacing strings.